public/model/OSMLayerFunctions.ts

```diff
@@ -46,12 +49,16 @@ const setLanguage = (maplibreRef: MaplibreRef, styleLayer: LayerSpecification) =
   }
 };
 
-const addNewLayer = (layerConfig: OSMLayerSpecification, maplibreRef: MaplibreRef) => {
+const addNewLayer = (
+  layerConfig: OSMLayerSpecification,
+  maplibreRef: MaplibreRef,
+  services: MapServices
+) => {
   if (maplibreRef.current) {
     const maplibre = maplibreRef.current;
     const { id, source, style } = layerConfig;
     addOSMLayerSource(maplibre, id, source.dataURL);
-    fetchStyleLayers(style?.styleURL).then((styleLayers: LayerSpecification[]) => {
+    fetchStyleLayers(style?.styleURL, services).then((styleLayers: LayerSpecification[]) => {
```

Comment:
Shall we catch the error here instead of passing `services` down to `fetchStyleLayers`? 

```typescript
fetchStyleLayers(style?.styleURL).then(...).catch(e => {
  services.toastNotifications.addWarning(RegionConflictWarningMsg);
})
```

Reply:
ack